.github/CONTRIBUTING.md

```diff
@@ -75,6 +75,114 @@ Also, have you signed the [Contribution License Agreement](https://cla.microsoft
 
 **If your contribution is going to be a major effort, you should give us a heads-up first. We have a lot of items captured in our backlog and we release every two weeks, so before you spend the time, just check with us to make sure your plans and ours are in sync :) Just open an issue on GitHub and tag it as "contribution".**
 
+#### Commit Guidelines
+We have very precise rules over how our git commit messages can be formatted. This leads to more readable messages that are easy to follow when looking through the project history.
+
+**Commit Message Format**
+
+Each commit message consists of a header, a body (optional) and a footer (optional). The header has a special format that includes a type, a scope and a subject:
+
+```java
```

Comment:
C#

Reply:
Fixed just a second before you posted :)